python/mcap/mcap0/data_stream.py

```diff
@@ -0,0 +1,99 @@
+import struct
+from io import BufferedIOBase, BytesIO
+
+from .exceptions import EndOfFile
+from .opcode import Opcode
+
+
+class ReadDataStream:
+    def __init__(self, stream: BufferedIOBase):
+        self.__count = 0
+        self.__stream = stream
+
+    def __del__(self):
+        self.__stream.close()
+
+    @property
+    def count(self) -> int:
+        return self.__count
+
+    def read(self, length: int) -> bytes:
+        if length == 0:
+            return b""
+
+        data = self.__stream.read(length)
+        self.__count += len(data)
+        if data == b"":
+            raise EndOfFile()
+        return data
+
+    def read1(self) -> int:
+        [value] = struct.unpack("<B", self.read(1))
```

Comment:
Docs say:

> Creating a `Struct` object once and calling its methods is more efficient than calling the `struct` functions with the same format since the format string only needs to be compiled once.

But also:
> Note: The compiled versions of the most recent format strings passed to Struct and the module-level functions are cached, so programs that use only a few format strings needn’t worry about reusing a single Struct instance.

I'm not sure what qualifies as "a few" though, so maybe it's better to use Struct instances?

Reply:
I would imagine we're in the "few" range here since there are < 10 we use but it might be interesting to do some benchmarking.